Update packaging to comply with standards

cmcantalupo opened the pull request.

dannosliwcd committed:
Clean up temporary files when PlatformTopoImp throws

dannosliwcd committed:
Reduce test log noise in MSRIOGroupTest

- Suppress warnings about unexpected mock calls to add_read and
  add_write in MSRIOGroupTest.
- Removes about 57 MB of noise from test logs

cmcantalupo committed:
Switch from setup.py to pyproject.toml

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Integrate new python build with autotools

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
[WIP] Remove python from make install target

- Manage python install in RPM or deb packaging
- Quick sketch on how to move python install, but
  none of the packaging works yet.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz commented:
It would be nice if we could integrate the sphinx docs build into the new build recipe.  It's difficult to deal with in spack when they're part of the regular autotools build, but I need to package geopmdpy separately (as a Python package) to resolve Python version issues between what's in spack vs. the host system.

cmcantalupo committed:
Use dynamic versioning for pyproject.toml files

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Move json data into python file and generate json files

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
fixup dynamic versioning

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Start to fixup debian packaging

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Build python sdist in autogen step to create version.py

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add schemas back to schemas directory

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Rely on python setuptools to get version from git repo

- Enables a single source of truth for version information
  now it is geopmdpy/version.py.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add manifest file for geopmdpy package

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Clean up autogen.sh script

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Rearrange everything

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo commented on service/setup.py:
```diff
@@ -1,61 +1,7 @@
-                    'psutil>=5.8.0',
-                    'dasbus>=1.6.0',
-                    'jsonschema>=3.2.0']
+import setuptools
```
comment why this is here (debian support)

cmcantalupo committed:
Add stub readme for geopmpy

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fixup some of the issues with reorg

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
[WIP] Adding individual .gitignore files

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Get autogen.sh script working

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Get libgeopmd build passing

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add a stub README for libgeopm directory

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Some clean up to get libgeopm build working

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fix symlinks in libgeopm/test directory

- Remove test_license.sh from make check target.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Collapse integration directories

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Remove copying_headers data, replaced by COPYING-TPP

- Move doxyen_update scripts into respective directories
- Delete TODO

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dannosliwcd commented on geopmdpy/requirements.txt:
Add `build` (used by autogen.sh)

dannosliwcd commented on geopmpy/requirements.txt:
Add `build` (used by autogen.sh)

cmcantalupo committed:
Remove links

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Get libgeopmd source dist working

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Remove sphinx from autogen.sh

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Hide pull request template

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
move tutorials under integration

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Clear out old "tutorial"

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
WIP debian files update for libgeopmd

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Generated files from debmake

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Move file copy lower in autogen.sh, after VERSION file is created

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Get make deb target working for libgeopmd

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Get debian packaging working for libgeopm

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Closer

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fixup libgeopm debian packaging

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Rename geopmdpy_test to just test

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
First attempt to provide debian packaging for geopmdpy

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add geopmdpy debian packaging

- get rid of version.py generation

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo commented on geopmdpy/geopmdpy/__init__.py:
```diff
@@ -1,26 +0,0 @@
-import os
-
-try:
-    from geopmdpy.version import __version__
```
i guess we may need to add this __init__ file back...

cmcantalupo commented on geopmdpy/debian/changelog:
```diff
@@ -1,5 +1,5 @@
-geopmdpy (3.0.1.dev222+gd00ee89a4.d20240328-1) UNRELEASED; urgency=low
+python3-geopmdpy (3.0.1.dev222+gd00ee89a4.d20240328-1) UNRELEASED; urgency=low
 
-  * Initial release. Closes: #1055130
```
closes line is handy

cmcantalupo committed:
Add symlinks from each package for repo files

- AUTHORS, COED_OF_CONDUCT.md CONTRIBUTING.rst COPYING and ChangeLog.
- Get version working with setuptools version 59.6
- Get the autotools builds working again
- Update copyright file

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dannosliwcd committed:
Add missing pygobject dependency to geopmdpy

dannosliwcd committed:
Update to new locations of msr json for TestMSRDataFiles

dannosliwcd committed:
Add installer to requirements.txt, needed for builds

dannosliwcd committed:
Make geopmdpy tests use the unittest discovery mechanism

dannosliwcd committed:
Update setuptools in requirements.txt for compatibility with setuptools_scm

dannosliwcd committed:
Add missing docstring-parser dependency

dannosliwcd committed:
Remove unused makefile from geopmdpy tests

dannosliwcd committed:
Use gtest runners for service tests

- Related to geopm#803
- Let the gtest unittest discovery routines find all of our
  test cases instead of manually registering each case in makefiles.
- Make gtest tests produce TAP logs.
- Update the make check target to consume test-anything protocol (TAP) logs.
- No longer need to manually maintain a list of test cases.

dannosliwcd committed:
Use gtest runners for libgeopm tests

    - Related to geopm#803
    - Let the gtest unittest discovery routines find all of our
      test cases instead of manually registering each case in makefiles.
    - Update the make check target to consume test-anything protocol
      (TAP) logs.

cmcantalupo committed:
Add packaging for geopmdpy

- Whitespace changes to the geopmdpy/debian/control file
- Prune requirements.txt

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz committed:
Add requirements file for dev builds

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Add unit test ignore files

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Sort ignore files

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo committed:
Add debmake generated copyright file for geopmpy

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
First draft of debian packaging for documentation

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz committed:
Purge python from geopmd RPM spec

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Purge man pages from geopmd RPM spec

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo committed:
Rename all of the C++ man pages

- Change all references to GEOPM_CXX_MAN

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz committed:
Bring back COPYING-TPP to dist

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Add build to base requirements.txt

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Use setuptools-scm version from python 3.6

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo committed:
Add docs makefile

- Use manpages files to install man pages

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz committed:
Draft geopmdpy.spec

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo committed:
Cleanup Makefile

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add a script to create all packages

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dannosliwcd committed:
Skip the PolicyStore python tests if libgeopm does not contain policystore symbols

dannosliwcd committed:
Resolve other test CFFI instances leaking into Endpoint tests

dannosliwcd committed:
Make geopmpy tests use the unittest discovery mechanism

dannosliwcd committed:
Add build-aux to the libgeopm .gitignore

dannosliwcd committed:
Update build paths in GH PR actions

dannosliwcd committed:
Fix include guards in libgeopmd mocks

dannosliwcd committed:
Set the python test return code based on pass/fail state

cmcantalupo committed:
Fix unit test when GPUs are present on the system.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dannosliwcd committed:
Add README descriptions to run test suites

dannosliwcd committed:
Only run geopm python tests without asan in GH actions

- Alternatively, we should force ASAN to run in shared-libasan mode

bgeltz committed:
Continue geopmdpy.spec draft

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

dannosliwcd committed:
Disable link-time optimization in debug builds

- Resolves #3308

dannosliwcd committed:
Update paths in codeql-analysis.yml

dannosliwcd committed:
Add build to geopmdpy .gitignore

dannosliwcd committed:
Add setuptools-scm to requirements.txt for geopmdpy and geopmpy

bgeltz committed:
WIP geopmdpy packaging

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Get geomdpy RPMs building

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

dannosliwcd committed:
Use generated version files

bgeltz committed:
Update make_rpm to use make_sdist

- Move version and archive settings to helper script;
  required for OBS.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo committed:
Use VERSION file when available for geopmdpy

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Remove dependency on top level script

- Clean up log files from sdist
- Introduce a python script specifically for creating an sdist

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fixup

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz committed:
Update make_deb to use make_sdist

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo committed:
Update version tooling for geopmpy

- Clean up .gitignore files

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Update debian changelog when packaging python

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Get docs versioning working

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Install systemd files in the right directory

- Remove sphinx requirements

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Clean up some requirements

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Remove python and bash completion from libgeopm dir build

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add spec file and make script for geopmpy

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
break things

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Revert "break things"

This reverts commit e1784c9920d0a4458a8626ab2cb8c07da29303d7.

cmcantalupo committed:
Some clean up for the geopmdpy spec file

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Move some scripting into docs/Makefile

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Include output of sphinx build in distribution tarball

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Get RPM docs building

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Clean up debian docs build

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add output spec files to .gitignore

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Remove dated steps from geopm-runtime.spec.in

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Small clean up to build and make

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Rename integration Makefile.mk to Makefile.am

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
First attempt at autotooling integration directory

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Remove integration prefix from all Makefiles

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Bring in most of libgeopm/configure.ac

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add 0.0.0 version file in case of pip install from github

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Work around for scm version issues

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Get rid of VERSION file for geopmdpy

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Do not generate VERSION file

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fall back to version 0.0.0 if package not installed

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add link to libgeopm/m4 dir

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
remove integration/ prefix from make files (again)

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Change check_PROGRAMS to noinst_PROGRAMS

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Move installed header files into include subdirectory

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Update configure to link and include libgeopm

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
fix more linking and include issues

- Update .gitignore

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Remove update_version.sh script

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Move shell completion files into docs directory

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add packaging for bash completion scripts

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Remove bash completion scripts from geopm-service install

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz committed:
Add libgeopm path to AM_FCFLAGS

- Use it in the nasft build.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Restore --disable-geopmd-local

- libgeopm is now assumed to be available
  either in the build tree, or in the
  user's environment.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Move *ModelRegion to libgeopm

- Was previously build into geopmbench.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo committed:
fixup header references in the Makefile.am for libgeopm

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
remove bash completion requirement from geopm-service.spec.in

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fixup includes in geopm-runtime.spec.in

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz committed:
Get build.sh working

- Add install/remove steps for geopmpy, geopmdpy.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo committed:
Add documentation packages as recommends for debian

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Protect openmp calls with ifdef

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
WIP rework top level README

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz committed:
Try to uninstall geopmdpy in build.sh

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Purge old Python pkg dir in run_env.sh

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

dannosliwcd committed:
Include geopm_version.h from geopm_version.c

bgeltz committed:
Add GEOPM_SOURCE to PYTHONPATH in run_env.sh

- Update path to service for version check.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Update package imports for new PYTHONPATH in run_env.sh

- Only GEOPM_SOURCE is added to PYTHONPATH now.
- All imports in subdirs should include 'integration'.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Purge test_tutorial_base

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

dannosliwcd committed:
Continue the top-level README work in progress

bgeltz committed:
amg: fix patch file context so it applies cleanly

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
hpcg: fix patch file context so it applies cleanly

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
hpl_netlib: restore qopenmp

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
minife: restore patch context so it applies cleanly

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
pennant: restore qopenmp and proper context

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
nekbone: restore qopenmp and context

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Get the experiment infrastructure working

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo committed:
Commit the C++ data files to the repo and modify the update_data.sh script to generate them

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Clean up C++ data generation script

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Some edits to the new base README

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Some more updates to the README

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fixup docs

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add force to clean of uninstalled man pages

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Simplify note about dependency packages and refer to build.html

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
fix requirements.txt permissions

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Simplify simple install script

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fixup

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Change two README.rst files to README.md

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Remove one more line from README script

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dannosliwcd committed:
Update READMEs for docs and shell_completion

cmcantalupo committed:
Add links to terminal videos

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dannosliwcd committed:
Update the builds for IOGroup and Agent tutorials

dannosliwcd committed:
Update non-root-directory READMEs

cmcantalupo committed:
Try to link to raw content

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Different links

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Update build documentation

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Update build and devel documentation

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Move doxygen generation into docs dir

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fixup rst in md file

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Clean up doxygen output

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add github scripting to upload packages to OBS

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dannosliwcd committed:
Use relative paths to doxygen and runtime documentation

bgeltz committed:
Correct python3-setuptools-scm name

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Refactor rpm/deb/sdist scripts for python packages

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Call make_sdist for python packages in OBS

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
